Packs/MISP/Integrations/MISPV3/MISPV3.py

```diff
@@ -884,7 +888,14 @@ def prepare_args_to_search(controller):
         args_to_misp_format['limit'] = '50'
     if 'tags' in args_to_misp_format:
         args_to_misp_format['tags'] = build_misp_complex_filter(args_to_misp_format['tags'])
-    args_to_misp_format['controller'] = controller
+    if 'with_attachments' in args_to_misp_format:
+        if demisto_args.get('with_attachments') == 'true':
+            args_to_misp_format['with_attachments'] = 1
+        else:
+            args_to_misp_format['with_attachments'] = 0
+            args_to_misp_format['controller'] = controller
```

Comment:
Is the `controller` depends on the `with_attachments`?

Reply:
Yes, the controller is required for the PyMISP Lib search. To get files from MISP you cannot use the artifact or event controller. Therefore i don't set the controller for the attachment pulling in the first palce. After the a file got downloaded it will basically rerun the function with the controller set to also get actual artifact details from MISP.